frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx

```diff
@@ -79,27 +80,45 @@ export const ActionParameterMappingForm = ({
 
   return (
     <div>
-      {actionParameters.map((actionParam: WritebackParameter) => (
-        <ParameterFormSection key={actionParam.id}>
-          <ParameterFormLabel>
-            {actionParam.name ?? actionParam.id}
-          </ParameterFormLabel>
-          <Select
-            value={currentMappings[getTargetKey(actionParam)] ?? null}
-            onChange={(e: SelectChangeEvent<string>) =>
-              handleParameterChange(e?.target?.value, actionParam.target)
-            }
-            options={[
-              { name: t`Ask the user`, value: null },
-              ...dashboardParameters.map(dashboardParam => ({
-                key: dashboardParam.id,
-                name: dashboardParam.name,
-                value: dashboardParam.id,
-              })),
-            ]}
-          />
-        </ParameterFormSection>
-      ))}
+      {actionParameters.map((actionParam: WritebackParameter) => {
+        const isHidden =
+          !!action?.visualization_settings?.fields?.[actionParam.id]?.hidden;
```

Comment:
4 optional chains is an obvious code smell and generates questions:
1. Why can an `action` be `undefined` in a `ActionParameterMapper`? If it can't, we should make code communicate this. For example we could do `if (typeof action === "undefined") { throw new Error("..."); }` in this component.
2. Are there any other places where we need to get so deep? Perhaps there are some utilities that could be reused and/or refactored to reduce the clutter. How about [`useActionForm`](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/actions/hooks/use-action-form/use-action-form.ts#L25)?

Reply:
I simplified expressions with introducing two separate helpers to get required and hidden states of the parameter.

As for general improvement of action existence, I'm not confident enough to make such refactoring right now

Reply:
actually you were right and I improved experience by reworking the component and props, thanks!